### ISSUE TITLE
bug: some icons appear jagged in safari/webkit browsers

### ISSUE THREAD

**Issue by danieletulone** [ionic-team/ionicons#1176]
### Current Behavior

<img width="375" alt="image" src="https://user-images.githubusercontent.com/35412801/216656548-bcfa0f12-dd27-4336-8805-11df78f6f8be.png">

### Expected Behavior

To not have rendering issues.

### Steps to Reproduce

- Take an iphone8;
- Go to ionicons website;
- Search for some icons (ex. people);
- Take a look!

**Comment by liamdebeasi** [comment]
Thanks for the report. What version of iOS are you testing this on?

**Comment by danieletulone** [comment]
I have tested on two iphone devices:
- iPhone 8, iOS version 16.0
- iPhone 11, iOS 16.1.2

**Comment by liamdebeasi** [comment]
Thanks for the additional information. Could you please clarify what the issue is with the rendering of the icons?

**Comment by danieletulone** [comment]
Of course!
As you can see the icons have some gliches.
![Screenshot (110)](https://user-images.githubusercontent.com/35412801/217383458-38de3b02-3d34-42dd-b799-91b70d9f0513.png)

**Comment by danieletulone** [comment]
Of course!
As you can see the icons have some gliches.
![image](https://user-images.githubusercontent.com/35412801/217383764-8688d501-9973-4e2f-8c13-760e5b60b459.png)

**Comment by xnousnow** [comment]
Does that occurs on any other icons?

**Comment by hacketiwack** [comment]
Same on my Dell screen with Safari 16.3

![Icon Badly Rendered](https://user-images.githubusercontent.com/13329768/219423878-32937347-f0b2-48a8-bbef-489cfe8574a8.png)

**Comment by liamdebeasi** [comment]
Thanks everyone! I can reproduce this on my end.

**Comment by hacketiwack** [comment]
@lincolnthree your tricks works for me on all the icons I tested: search and language.
For those icons, I set the stroke attribute to inherit.
However, it doesn't work, for example, for the icon menu.

**Comment by matfantinel** [comment]
I am also facing this on an Ionic app, when running on iOS.

I did run some experiments and it seems to get fixed, at least sometimes, when adding `-webkit-transform: translate3d(0,0,0)` to the icons. From what I understood, this line of CSS makes some devices enable hardware acceleration on that element, which might result in a better rendering.

In my ionic-angular app, I added to the global.scss file:

```css
ion-icon {
  -webkit-transform: translate3d(0,0,0);
}
```

I styled the ion-icon element itself instead of the svg directly because I can't access it because of the Shadow DOM.

**Comment by fer0n** [comment]
> For those icons, I set the stroke attribute to inherit.
> However, it doesn't work, for example, for the icon menu.

How were you able to do this? That stroke is set directly on the svg element of the `IonIcon`, which is using a shadow dom and there aren't any shadow parts exposed. Am I missing something?

**Comment by matfantinel** [comment]
@fer0n I think that's only possible if you use the SVGs directly instead of the IonIcon component. Might be worth a shot, if only to see if that fix actually works for you. Then the options would either be to use the SVGs directly whenever the icons have this issue or fork Ionicons and expose the shadow part.

Hopefully either Safari gets its $#1T together or the Ionic team figures out a better fix for this soon

**Comment by fer0n** [comment]
Thanks!
I ended up wrapping the `IonIcon` in a component that removes the stroke from the icon. Not ideal, but I don't see any good solutions.

```
import { IonIcon } from "@ionic/react";
import React, { useMemo } from "react";

interface IonIconProps {
  color?: string;
  flipRtl?: boolean;
  icon?: string;
  ios?: string;
  lazy?: boolean;
  md?: string;
  mode?: "ios" | "md";
  name?: string;
  size?: string;
  src?: string;
}

/**
 * **Do not use if icon needs a stroke!** Fixes jagged icon appearance on iOS for filled icons.
 */
export default function IonIconNoStroke({ icon, ...props }: IonIconProps) {
  const iconWithoutStrokeStyle = useMemo(() => {
    if (icon) {
      return removeStrokeStyleFromIcon(icon);
    }
  }, [icon]);

  return <IonIcon icon={iconWithoutStrokeStyle ?? icon} {...props} />;
}

function removeStrokeStyleFromIcon(icon: string) {
  const tempElement = document.createElement("div");
  tempElement.innerHTML = icon;
  const cleanedElement = tempElement.querySelector("svg");
  if (cleanedElement) {
    cleanedElement.style["stroke"] = "unset";
    return tempElement.innerHTML;
  }
}

```

**Comment by aeharding** [comment]
Unfortunately, this fix is much more hacky for projects using Ionic then if it were patched in Ionic directly.

@liamdebeasi would it be possible to either add `stroke: unset` to Ionic Icons package, or alternatively add `slot` so we can fix this with a CSS selector?

Thanks!!

**Comment by liamdebeasi** [comment]
It's the same solution as https://github.com/ionic-team/ionicons/issues/1176#issuecomment-1572685263, but the solution is applied to the `svg` inside of `ion-icon`, so it should work when using the `ion-icon` component too. The solution provided in the linked comment only worked when you were working with the SVG directly.

**Comment by matfantinel** [comment]
@liamdebeasi unfortunately it still has the same issue after installing that version of the package. Fixes on desktop Safari, but still looks choppy on iOS (I tested as an installed app with Capacitor)

**Comment by liamdebeasi** [comment]
Thanks for testing. Do you have a screenshot of the issue happening on iOS? I was able to reproduce some jaggedness all the way back to iOS 14, so it's possible there are multiple SVG rendering bugs at play.

**Comment by matfantinel** [comment]
@liamdebeasi here it is! It's really similar to the one earlier in this thread. This screenshot was after applying the tentative fix, but it looked identical before.



![IMG_2645](https://github.com/ionic-team/ionicons/assets/24247035/7fe458e5-f0be-4f21-8f78-e5f8267c39be)

**Comment by aeharding** [comment]
Attaching for reference:

https://bugs.webkit.org/show_bug.cgi?id=258375

**Comment by Beelink** [comment]
> Unfortunately, this fix is much more hacky for projects using Ionic then if it were patched in Ionic directly.
> 
> @liamdebeasi would it be possible to either add `stroke: unset` to Ionic Icons package, or alternatively add `slot` so we can fix this with a CSS selector?
> 
> Thanks!!

Looks like a perfect solution!

Any updates on this?

**Comment by aeharding** [comment]
This appears to be resolved in iOS 17.4
